packages/source-iottwinmaker/src/data-binding-provider/createEntityPropertyBindingProvider.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { createMockTwinMakerSDK } from '../__mocks__/iottwinmakerSDK';
+import { TwinMakerMetadataModule } from '../metadata-module/TwinMakerMetadataModule';
+import { createEntityPropertyBindingProvider } from './createEntityPropertyBindingProvider';
+
+describe('createEntityPropertyBindingProvider', () => {
+  const getEntity = jest.fn();
+  const listEntities = jest.fn();
+  const tmClient = createMockTwinMakerSDK({
+    getEntity,
+    listEntities,
+  });
+  const mockWorkspaceId = 'ws-id';
+  const metadataModule = new TwinMakerMetadataModule(mockWorkspaceId, tmClient);
+
+  const mockDataBindingInput = {
+    dataBindingContext: {
+      entityId: 'entity-option-1',
+      componentName: 'component-3',
+      propertyName: 'alarm_key',
+    },
+  };
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('should return a correct data binding provider store', async () => {
+    const provider = createEntityPropertyBindingProvider({ metadataModule, timeSeriesDataQuery: jest.fn() });
+    const store = provider.createStore(false);
+    expect(store).toBeDefined();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    expect((store as any)['metadataModule']).toBe(metadataModule);
```

Comment:
Can we do this without casting as any? Getting the correct type on store, using a type guard, or casting as the correct type, are all preferable solutions to casting as any. Could we try one of them?

Reply:
the return type of this function doesn't have `metadataModule` so I need to cast. but I can cast it to the class that implements the return type.